common/src/main/java/com/hedera/block/common/utils/ChunkUtils.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.common.utils;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class ChunkUtils {
+    public static <T> List<List<T>> chunkify(final Collection<T> collection, final int chunkSize) {
+        Objects.requireNonNull(collection);
+        if (chunkSize <= 0) {
+            throw new IllegalArgumentException("Chunk size must be greater than 0");
+        }
+        if (collection.isEmpty()) {
+            return Collections.emptyList(); // or throw, depends on how we want to handle
```

Comment:
Should we throw here or return an empty collection? What should be the general behavior?

Reply:
I prefer to return an empty collection, I think it makes it more generic to any use case.
Maybe just add to the javadoc the expected behaviour to keep things simple.

Reply:
Awesome, sounds good.

Reply:
✅  Done!